### PR TITLE
Extend search component

### PR DESCRIPTION
**Pull request by menkuichan**

### CODE REVIEW

File: src/components/Search/List.jsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+  return (
+    <ListWrapper>
+      {(movies.length)
+        ? movies.map((movie, index) => (
+          <Item
+            key={index} // eslint-disable-line
+          >
+            <Poster
+              src={`https://image.tmdb.org/t/p/w94_and_h141_bestv2${movie.poster_path}`}
```
Comment (ok2ju) [inline src/components/Search/List.jsx]: Move image base url into constants

File: src/components/Search/List.jsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    }
+    return '';
+  };
+
+  return (
+    <ListWrapper>
+      {(movies.length)
+        ? movies.map((movie, index) => (
+          <Item
+            key={index} // eslint-disable-line
```
Comment (ok2ju) [inline src/components/Search/List.jsx]: when you have only one prop you can write it on the same line with component name

File: src/components/Search/List.jsx [hunk 3]
```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {
+  ListWrapper, Item, Poster, Info, Title, Overview,
+} from './styles';
+
+const List = ({ movies = [] }) => {
+  const getYear = (date) => {
```
Comment (ok2ju) [inline src/components/Search/List.jsx]: This function don't depends on any props in component, so why you define it inside of `List` component? On every re-render of `List` component `getYear` function will be created in memory again and again.

File: src/components/Search/List.jsx [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    if (date) {
+      return `(${new Date(date).getFullYear()})`;
+    }
+    return '';
+  };
+
+  return (
+    <ListWrapper>
+      {(movies.length)
```
Comment (ok2ju) [inline src/components/Search/List.jsx]: Check if `length` > 0 or cast `length` to bool using `!!`. Also you don't need braces `()` here

File: src/components/Search/List.jsx [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+          >
+            <Poster
+              src={`https://image.tmdb.org/t/p/w94_and_h141_bestv2${movie.poster_path}`}
+            />
+            <Info>
+              <Title>
+                {`${movie.title} ${getYear(movie.release_date)}`}
+              </Title>
+              <Overview>
+                {`${movie.overview}`}
```
Comment (ok2ju) [inline src/components/Search/List.jsx]: Why do you need template string ``${}`` here?

File: src/components/Search/List.jsx [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+              </Overview>
+            </Info>
+          </Item>
+        ))
+        : <Item textColor="gray">No results</Item>}
```
Comment (ok2ju) [inline src/components/Search/List.jsx]: Don't use css attributes on components, move it in styles

File: src/components/Search/Search.jsx [hunk 7]
```diff
@@ -1,31 +1,44 @@
...
-  const onEnterPress = (e) => {
-    if (e.charCode === ENTER_KEY) {
-      setvalue(value);
+  const onHandleChange = async (e) => {
+    const currentValue = e.target.value;
+    setValue(currentValue);
+    if (!currentValue.length) {
+      setData([]);
+    }
+    if (currentValue.trim().length) {
```
Comment (ok2ju) [inline src/components/Search/Search.jsx]: check if `length` > 0 or cast to bool `!!`

File: src/components/Search/Search.jsx [hunk 8]
```diff
@@ -1,31 +1,44 @@
...
-    <SearchWrapper>
-      <SearchInput
-        placeholder="Search…"
-        onKeyPress={onEnterPress}
-        onChange={onHandleChange}
-        value={value}
-      />
-      <SearchIcon />
-    </SearchWrapper>
+    <div>
```
Comment (ok2ju) [inline src/components/Search/Search.jsx]: Do you need empty `<div>` here? If you need just any block here you can use `<React.Fragment>`

File: src/components/Search/Search.jsx [hunk 9]
```diff
@@ -1,31 +1,44 @@
...
+        <SearchInput
+          placeholder="Search…"
+          onChange={onHandleChange}
+          value={value}
+        />
+        <IconContainer>
+          <SearchIcon />
+        </IconContainer>
+      </SearchWrapper>
+      {value.length ? <List movies={data} /> : <div />}
```
Comment (ok2ju) [inline src/components/Search/Search.jsx]: Check if `length` > 0 or cast to bool `!!`. Why do you need to render empty `<div>` here? You can rewrite this expression using binary operator, e.g.:
```js
{!!value.length && <List movies={data} />}
```

File: src/components/Search/search.stories.js [hunk 10]
```diff
@@ -0,0 +1,6 @@
+import React from 'react';
+import { storiesOf } from '@storybook/react'; // eslint-disable-line
+import Search from '.';
+
+storiesOf('Search', module).add('default',
```
Comment (ok2ju) [inline src/components/Search/search.stories.js]: Please, fix code formatting

File: src/components/Search/styles.js [hunk 11]
```diff
@@ -1,20 +1,99 @@
...
   font-size: 15px;
-  background-color: ${(props) => props.theme.main.color};
+  background-color: transparent;
+  padding: 6px 15px;
+  border: 1px solid #716F73;
+  border-radius: 5px;
 
   &:focus {
     outline: none;
+    background-color: white;
```
Comment (ok2ju) [inline src/components/Search/styles.js:19]: write color as hex code

File: src/components/Search/styles.js [hunk 12]
```diff
@@ -1,20 +1,99 @@
...
+`;
+
+export const Item = styled.div`
+  display: flex;
+  flex-direction: row;
+  justify-content: space-around;
+  align-items: flex-start;
+  width: 230px;
+  padding: 10px;
+  color: ${(props) => props.textColor};
```
Comment (ok2ju) [inline src/components/Search/styles.js]: grab this color from theme

File: src/constants.js [hunk 13]
```diff
@@ -2,4 +2,12 @@ export const NAV_LINKS = [
   'Popular', 'Now playing', 'Upcoming', 'Most rated',
 ];
 
-export const ENTER_KEY = 13;
+const PORT = 3000;
+
+export const API_URL = `http://localhost:${PORT}`;
```
Comment (ok2ju) [inline src/constants.js:5]: Where do you use this constant?

File: src/constants.js [hunk 14]
```diff
@@ -2,4 +2,12 @@ export const NAV_LINKS = [
...
+
+export const LIST_PAGE = 1;
```
Comment (ok2ju) [inline src/constants.js]: Name this constant more specific. It uses only for `Search`

File: src/constants.js [hunk 15]
```diff
@@ -2,4 +2,12 @@ export const NAV_LINKS = [
...
+
+export const LIST_PER_PAGE = 3;
```
Comment (ok2ju) [inline src/constants.js]: Same as above

File: src/index.jsx [hunk 16]
```diff
@@ -1,11 +1,18 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
-import { ThemeProvider } from 'styled-components';
+import { ThemeProvider, createGlobalStyle } from 'styled-components';
 import MoviesPage from './Pages/MoviePage';
-import { theme } from './theme';
+import theme from './theme';
+
+const GlobalStyle = createGlobalStyle`
```
Comment (ok2ju) [inline src/index.jsx]: Move this into separate component `GlobalStyles` in `components` folder

File: webpack.config.js [hunk 17]
```diff
@@ -1,6 +1,7 @@
 const HtmlWebPackPlugin = require('html-webpack-plugin');
 
 module.exports = {
+  entry: ['@babel/polyfill', './src/index.jsx'],
```
Comment (ok2ju) [inline webpack.config.js:4]: Why do you need this polyfill?

File: src/theme.js [hunk 18]
```diff
@@ -1,7 +1,7 @@
-export const theme = {
+export default {
   main: {
     color: '#110F10',
     orange: '#f79b0e',
-    gray: '#f2f2f2',
+    gray: '#808080',
```
Comment (ok2ju) [inline src/theme.js]: strange theme color names. i suggest you use more specific names, e.g.
```js
search: {
  placeholder: '#808080`,
  label: '#ccc',
  ...
}
```

File: src/components/Icons/SearchIcon.jsx [hunk 19]
```diff
@@ -4,7 +4,7 @@ import PropTypes from 'prop-types';
 const SearchIcon = ({
   fill, width, height, viewBox,
 }) => (
-  <svg width={width} height={height} viewBox={viewBox} fill={fill} xmlns="http://www.w3.org/2000/svg">
+  <svg style={{ display: 'block' }} width={width} height={height} viewBox={viewBox} fill={fill} xmlns="http://www.w3.org/2000/svg">
```
Comment (ok2ju) [inline src/components/Icons/SearchIcon.jsx]: No, you shouldn't write such styles here. You can apply this styles in `IconContainer` styled-component like this:
```js
export const IconContainer = styled...`
  position: relative;
  ...

  svg {
    display: block;
  }
`
```

**Review by ok2ju — CHANGES_REQUESTED** [review]

File: src/constants.js (hunk 13 above)
Comment (menkuichan) [reply src/constants.js:5]: At ```src/api/config.js``` for axios instanse.

File: src/constants.js (hunk 14 above)
Comment (menkuichan) [reply src/constants.js]: May it be```SEARCH_LIST_PAGE```?

File: webpack.config.js (hunk 17 above)
Comment (menkuichan) [reply webpack.config.js:4]: Without ```@babel/polyfill```, a "regeneratorRuntime is not defined" error occurs when trying to use async/await.

File: src/components/Header/Header.jsx [hunk 20]
```diff
@@ -6,7 +6,7 @@ const Header = styled.header`
   padding: 0 20px 0 20px;
   display: flex;
   justify-content: space-between;
-  background-color: ${(props) => props.theme.main.color};
+  background-color: ${({ theme: { header } }) => header.background};
```
Comment (ok2ju) [inline src/components/Header/Header.jsx]: I don't like deep destructuring, please use one level destructurisation, e.g. `${({ theme }) => ...}

File: src/components/Nav/styles.js [hunk 21]
```diff
@@ -5,7 +5,7 @@ export const LinkButton = styled.button`
   background: none;
   font-size: 15px;
   padding: 20px 30px;
-  color: ${({ index, value, theme }) => (index === value ? theme.main.orange : theme.main.gray)};
+  color: ${({ index, value, theme: { header } }) => (index === value ? header.activeLink : header.text)};
```
Comment (ok2ju) [inline src/components/Nav/styles.js]: Same as above

File: src/components/Search/Search.jsx [hunk 22]
```diff
@@ -1,30 +1,48 @@
...
+import { SEARCH_PARAMS } from '../../constants';
+import {
+  SearchWrapper, SearchInput, IconContainer, InputContainer,
+} from './styles';
 
 const Search = () => {
-  const [value, setvalue] = useState('');
+  const [value, setValue] = useState('');
+  const [data, setData] = useState([]);
+  const debouncedValue = useDebounce(value, 1000);
```
Comment (ok2ju) [inline src/components/Search/Search.jsx]: 1 sec is too long, let's change it to `200ms`

File: src/components/Search/Search.jsx [hunk 23]
```diff
@@ -1,30 +1,48 @@
...
+    } else {
+      setData([]);
     }
-  };
+  }, [debouncedValue]);
 
-  const onHandleChange = (e) => {
-    setvalue(e.target.value);
+  const onHandleChange = (event) => {
+    const { value: currValue } = event.target;
```
Comment (ok2ju) [inline src/components/Search/Search.jsx]: Why do you need destructuring here? You don't use this variable more than once.

File: src/components/Search/styles.js [hunk 24]
```diff
@@ -1,20 +1,113 @@
 import styled from 'styled-components';
+import defaultTheme from '../../theme';
+
+const inputHeight = 31;
 
 export const SearchInput = styled.input`
   border: 0;
-  width: 200px;
-  color: ${(props) => props.theme.main.gray};
+  color: ${({ theme: { header } }) => header.text};
```
Comment (ok2ju) [inline src/components/Search/styles.js]: same as above

**Review by ok2ju — CHANGES_REQUESTED** [review]

**Review by ok2ju — APPROVED** [review]

**Pull request merged**